mezzaluna-feature-loader/src/loaded_features/mod.rs

```diff
@@ -231,29 +161,63 @@ impl<S> LoadedFeatures<S> {
 
     /// An iterator visiting all features in arbitrary order. The iterator
     /// element type is `&'a Feature`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mezzaluna_feature_loader::{Feature, LoadedFeatures};
+    ///
+    /// let mut features = LoadedFeatures::new();
+    /// features.insert(Feature::with_in_memory_path("/src/_lib/test.rb".into()));
+    /// features.insert(Feature::with_in_memory_path("set.rb".into()));
+    /// features.insert(Feature::with_in_memory_path("artichoke.rb".into()));
+    ///
+    /// for f in features.features() {
+    ///     println!("Loaded feature at: {}", f.path().display());
+    /// }
+    /// ```
     #[must_use]
     pub fn features(&self) -> Features<'_> {
         let inner = self.features.iter();
         Features { inner }
     }
 
     /// Returns the number of features in the set.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mezzaluna_feature_loader::{Feature, LoadedFeatures};
+    ///
+    /// let mut features = LoadedFeatures::new();
+    /// assert_eq!(features.len(), 0);
+    ///
+    /// features.insert(Feature::with_in_memory_path("/src/_lib/test.rb".into()));
+    /// assert_eq!(features.len(), 1);
+    /// ```
     #[must_use]
     pub fn len(&self) -> usize {
         self.features.len()
     }
 
     /// Returns true if the set contains no features.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use mezzaluna_feature_loader::{Feature, LoadedFeatures};
+    ///
+    /// let mut features = LoadedFeatures::new();
+    /// assert!(features.is_empty());
+    ///
+    /// features.insert(Feature::with_in_memory_path("/src/_lib/test.rb".into()));
+    /// assert!(!features.is_empty());
+    /// ```
     #[must_use]
     pub fn is_empty(&self) -> bool {
         self.features.is_empty()
     }
 
-    /// Clears the set, removing all values.
-    pub fn clear(&mut self) {
```

Comment:
this API violated the API contract of this container that it should be append-only.